File: composeApp/src/androidMain/kotlin/com/humblesolutions/indsphinx/repository/BackendCoordinatorFormRepository.kt
package com.humblesolutions.indsphinx.repository

import com.google.firebase.firestore.FieldValue
import com.google.firebase.firestore.FirebaseFirestore
import com.google.firebase.firestore.Query
import com.humblesolutions.indsphinx.model.MonthlyCheckForm
import kotlinx.coroutines.tasks.await

class BackendCoordinatorFormRepository : CoordinatorFormRepository {
    private val db = FirebaseFirestore.getInstance()

    override suspend fun submitForm(form: MonthlyCheckForm) {
        val data = mapOf(
            "occupantId" to form.occupantId,
            "flatId" to form.flatId,
            "occupantName" to form.occupantName,
            "flatNumber" to form.flatNumber,
            "month" to form.month,
            "cleanliness" to form.cleanliness,
            "repairs" to form.repairs,
            "safety" to form.safety,
            "bills" to form.bills,
            "hrIssues" to form.hrIssues,
            "confirmed" to form.confirmed,
            "submittedAt" to FieldValue.serverTimestamp()
        )
        db.collection("forms").add(data).await()
    }

    override suspend fun getLastFormSubmittedAt(occupantId: String): Long? {
        val query = db.collection("forms")
            .whereEqualTo("occupantId", occupantId)
            .orderBy("submittedAt", Query.Direction.DESCENDING)
            .limit(1)
            .get().await()
        val ts = query.documents.firstOrNull()?.getTimestamp("submittedAt") ?: return null
        return ts.toDate().time
    }
}

File: iosApp/iosApp/BackendCoordinatorFormRepository.swift
import Foundation
import FirebaseFirestore

class BackendCoordinatorFormRepository {
    private let db = Firestore.firestore()

    func submitForm(
        occupantId: String,
        flatId: String,
        occupantName: String,
        flatNumber: String,
        month: String,
        cleanliness: [String: String],
        repairs: [String: String],
        safety: [String: String],
        bills: [String: String],
        hrIssues: String
    ) async throws {
        let data: [String: Any] = [
            "occupantId": occupantId,
            "flatId": flatId,
            "occupantName": occupantName,
            "flatNumber": flatNumber,
            "month": month,
            "cleanliness": cleanliness,
            "repairs": repairs,
            "safety": safety,
            "bills": bills,
            "hrIssues": hrIssues,
            "confirmed": true,
            "submittedAt": FieldValue.serverTimestamp()
        ]
        try await db.collection("forms").addDocument(data: data)
    }
}
